core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -777,10 +807,128 @@ NavigableMap<Long, InFlightBatch> cachedState() {
+                            }
+                        } else {
+                            log.debug("The batch is not in acquired state while release of acquisition lock on timeout, skipping, batch: {}"
+                                    + " for the share group: {}-{}-{}", inFlightBatch, groupId, memberId, topicIdPartition);
+                        }
+                    } else { // Case when batch has a valid offset state map.
+                        for (Map.Entry<Long, InFlightState> offsetState : inFlightBatch.offsetState.entrySet()) {
+                            // For the first batch which might have offsets prior to the request base
+                            // offset i.e. cached batch of 10-14 offsets and request batch of 12-13.
+                            if (offsetState.getKey() < baseOffset) {
+                                continue;
+                            }
+
+                            if (offsetState.getKey() > lastOffset) {
+                                // No further offsets to process.
+                                break;
+                            }
+
+                            if (offsetState.getValue().state != RecordState.ACQUIRED) {
+                                log.debug("The offset is not in acquired state while release of acquisition lock on timeout, skipping, offset: {} batch: {}"
+                                                + " for the share group: {}-{}-{}", offsetState.getKey(), inFlightBatch,
+                                        groupId, memberId, topicIdPartition);
+                                continue;
+                            }
+                            InFlightState updateResult = offsetState.getValue().tryUpdateState(RecordState.AVAILABLE, false);
+                            if (updateResult == null) {
+                                log.debug("Unable to release acquisition lock on timeout for the offset: {} in batch: {}"
+                                                + " for the share group: {}-{}-{}", offsetState.getKey(), inFlightBatch,
+                                        groupId, memberId, topicIdPartition);
+                                continue;
+                            }
+                            // Update acquisition lock timeout task for the offset to null since it is completed now.
+                            updateResult.updateAcquisitionLockTimeoutTask(null);
+                            localNextFetchOffset = Math.min(offsetState.getKey(), localNextFetchOffset);
+                        }
+                    }
+                }
+                nextFetchOffset = localNextFetchOffset;
+            }
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
     /**
      * The InFlightBatch maintains the in-memory state of the fetched records i.e. in-flight records.
      */
-    static class InFlightBatch {
+    class InFlightBatch {
```

Comment:
Why do we want to change this?

Reply:
In the function `maybeInitializeOffsetStateUpdate()`, I am doing `timer.add(timerTask)`. `timer` is non-static and in outer class `SharePartition`, so it won't be able to access it. In case I had to stick with `static class InFlightBatch`, I would have to make a bunch of functions and `timer` as static in the outer class `SharePartition`